core/src/main/scala/kafka/coordinator/transaction/TransactionCoordinator.scala

```diff
@@ -181,6 +183,7 @@ class TransactionCoordinator(txnConfig: TransactionConfig,
               newMetadata.producerEpoch,
               TransactionResult.ABORT,
               isFromClient = false,
+              clientTransactionVersion = 0,
```

Comment:
In the fencing case, we have separate logic for bumping the epoch. It does something slightly different and I didn't want to change that logic in this PR.

Reply:
I was thinking about this some more. We can support setting 0 here and only using the new epoch bump logic in the non prepareProducerFence/fromClient cases. But I also wonder, since these are not even initiated by the client, does the naming of this value even make sense? I do see that maybe we would want to use the tagged fields and extra features, so I will probably just use the server-side TV. But I am wondering if this name will be confusing.